### ISSUE TITLE
[BUG] check_pdmultiindex_panel incorrect result for metadata is_equally_spaced

### ISSUE THREAD

**Issue by noahleegithub** [sktime/sktime#4299]
**Describe the bug**

The *check_pdmultiindex_panel* method from sktime.datatypes will return an incorrect result for the "is_equally_spaced" metadata when the time index for each instance has a non-sequential initial time.

**To Reproduce**

```python
import pandas as pd
from sktime.datatypes import check_is_mtype

example_instance_index = [0, 0, 0, 1, 1, 1, 2, 2, 2]
example_time_index = pd.to_datetime([0, 1, 2, 4, 5, 6, 9, 10, 11], unit='s')
example_multiindex = pd.MultiIndex.from_arrays([example_instance_index, example_time_index])

example_data = pd.DataFrame(index=example_multiindex, data=list(range(0, 9)))
"""example_data
                       0
0 1970-01-01 00:00:00  0
  1970-01-01 00:00:01  1
  1970-01-01 00:00:02  2
1 1970-01-01 00:00:04  3
  1970-01-01 00:00:05  4
  1970-01-01 00:00:06  5
2 1970-01-01 00:00:09  6
  1970-01-01 00:00:10  7
  1970-01-01 00:00:11  8
"""

_, _, metadata = check_is_mtype(example_data, mtype="pd-multiindex", return_metadata=True)

assert metadata['is_equally_spaced'] == True
```

**Expected behavior**

The method should return True for "is_equally_spaced", since for each separate instance in the DataFrame the spacing between time indexes is 1 second.

**Additional context**

In the check_pdmultiindex_panel method (sktime/datatypes/_panel/_check.py) the is_equally_spaced metadata is computed via
```python
metadata["is_equally_spaced"] = all(
            _index_equally_spaced(group.index.levels[-1]) for _, group in series_groups
        )
```
If the get_level_values() method is used instead of the levels attribute, this bug should be fixed.
```python
metadata["is_equally_spaced"] = all(
            _index_equally_spaced(group.index.get_level_values(-1)) for _, group in series_groups
        )
```

**Versions**
<details>

System:
    python: 3.9.15 | packaged by conda-forge | (main, Nov 22 2022, 15:55:03)  [GCC 10.4.0]
executable: /home/bin/python
   machine: Linux-4.14.305-10ms-x86_64-with-glibc2.27

Python dependencies:
          pip: 23.0.1
   setuptools: 58.1.0
      sklearn: 1.2.1
       sktime: 0.16.1
  statsmodels: 0.13.5
        numpy: 1.23.5
        scipy: 1.10.1
       pandas: 1.5.3
   matplotlib: None
       joblib: 1.2.0
        numba: 0.56.4
     pmdarima: None
      tsfresh: 0.20.0
</details>

**Comment by fkiraly** [comment]
yes, can confirm the bug on windows, python 3.10, current `main`.

I had to think about the "meaning" of `is_equally_spaced` - i.e., whether it should mean "equal index across instances, and also equally spaced". I was not sure whether that is tacitly assumed by estimators, but in the end I feel we should go with the description of the tag.

Also thanks for the proposed fix, this seems to work.

Would appreciate a PR!
Could you also add your test as a single pytest test at the end of `datatypes.tests.test_check`?

**Comment by noahleegithub** [comment]
@fkiraly Yes, I can make a PR for this. I've also noticed that a very similar issue exists for the convert method `from_multi_index_to_3d_numpy`, i.e. the bug described in https://github.com/sktime/sktime/issues/3536. Right now, the workaround is to either replace the time index with a `RangeIndex`, which is what the `PaddingTransformer` does, or zero out the time indices for each instance separately. Right now, the method performs the conversion by assuming that each instance will contain the exact same time indices, which causes the numpy shape error. Do you think this method should be updated to allow each instance to contain different time indices, while validating that each instance has equal time series length and spacing?

**Comment by fkiraly** [comment]
Hmm, thanks for pointing this out, @noahleegithub, and thanks for the fix!

I think currently there are a lot of places in the codebase where "equal index across instances" is tacitly assumed.

There are two things in play in #3536 imo:

* For `from_multi_index_to_3d_numpy`, that seems fine, this should only work if the index is equal, and equally spaced - as `numpy3d` is by default interpreted as a panel of equal length, equal spaced time series.
* For transformers, on the other hand, it is worthwhile considering whether the internal logic can be replace by sth faster or closer to the `pandas` dtypes. We would like to get away from `nested_univ` as it is a bit dodgy since `pandas` 1.5.0. `numpy` can be more efficient than `pandas`,  but `pandas` allows unequal indices, and is closer to `dask`.
